common/rfb/pam.c

```diff
@@ -84,8 +87,22 @@ int do_pam_auth(const char *service, const char *username, const char *password)
   };
   pam_handle_t *h = 0;
   ret = pam_start(service, username, &conv, &h);
-  if (ret == PAM_SUCCESS)
+  if (ret == PAM_SUCCESS) {
+
+#ifdef PAM_XDISPLAY
+    /* Pass the display name to PAM modules. */
+    ret = pam_set_item(h, PAM_XDISPLAY, displayStr);
+    /* Note: PAM_XDISPLAY is only supported by modern versions of PAM */
```

Comment:
It's almost 20 years old. Do such systems still exist, and will run binaries built on a more modern system?

Reply:
It's tested in [vncsession.c](https://github.com/TigerVNC/tigervnc/blob/5bfcb7a0c7559c3f029da68d89619e8944021fa8/unix/vncserver/vncsession.c#L215). I just followed the pattern. Anyway, the test is harmless.

Reply:
I confirmed that PAM_XDISPLAY does not exist in Solaris 11.4, so the test is still necessary.